elytron/src/main/java/org/wildfly/extension/elytron/ModifiableRealmDecorator.java

```diff
@@ -503,31 +537,42 @@ protected void executeRuntimeStep(final OperationContext context, final ModelNod
         }
 
         private Password createPassword(final OperationContext parentContext, final String principalName, String passwordType, ModelNode passwordNode) throws OperationFailedException, NoSuchAlgorithmException, InvalidKeySpecException {
-            String password = SetPasswordHandler.Bcrypt.PASSWORD.resolveModelAttribute(parentContext, passwordNode).asString();
+            String password;
             final PasswordSpec passwordSpec;
             final String algorithm;
 
             if (passwordType.equals(ElytronDescriptionConstants.BCRYPT)) {
+                password = SetPasswordHandler.Bcrypt.PASSWORD.resolveModelAttribute(parentContext, passwordNode).asString();
                 byte[] salt = SetPasswordHandler.Bcrypt.SALT.resolveModelAttribute(parentContext, passwordNode).asBytes();
                 int iterationCount = SetPasswordHandler.Bcrypt.ITERATION_COUNT.resolveModelAttribute(parentContext, passwordNode).asInt();
                 passwordSpec = new EncryptablePasswordSpec(password.toCharArray(), new IteratedSaltedPasswordAlgorithmSpec(iterationCount, salt));
                 algorithm = SetPasswordHandler.Bcrypt.ALGORITHM.resolveModelAttribute(parentContext, passwordNode).asString();
             } else if (passwordType.equals(ElytronDescriptionConstants.CLEAR)) {
+                password = SetPasswordHandler.Bcrypt.PASSWORD.resolveModelAttribute(parentContext, passwordNode).asString();
```

Comment:
You're right Brian; this has been a complete oversight on my part. Right now should be moved to SetPasswordHandler level because OTP has no password attribute. We can compose them extending from a base class but maybe just for a single attribute heredity, maybe it doesn't make sense, WDYT? Anyway there is a slight difference between password definitions on each attribute, Bcrypt, Clear and SimpleDigest have a minimum size defined, SaltedSimpleDigest and Digest don't. Which has no so much sense to me, but I will clear it up.